script.js

```diff
@@ -0,0 +1,980 @@
+      }
+
+      if ((keyCode === 'ShiftRight') || (keyCode === 'ShiftLeft')) {
+        this.shift = true;
+        this.changeWordCase();
+      }
+    }
+  }
+
+  mouseUpEvent() {
+    const q = document.querySelectorAll('.active');
+
+    q.forEach((item) => {
+      if (!item.classList.contains('CapsLock')) {
+        item.classList.remove('active');
+      }
+    });
+
+    this.shift = false;
+    this.changeWordCase();
+  }
+
+  keyDownEvent(event) {
+    const codeKey = event.code;
+    const button = document.querySelector(`.${codeKey}`);
+
+    if (button) {
+      event.preventDefault();
+
+      button.classList.add('active');
+      button.click();
+
+      if ((codeKey === 'AltLeft') || (codeKey === 'AltRight')) {
+        this.alt = true;
+      }
+
+      if ((codeKey === 'ShiftRight') || (codeKey === 'ShiftLeft')) {
+        this.shift = true;
+      }
+
+      this.changeWordCase();
+    }
+  }
+
+  keyUpEvent(event) {
+    const codeKey = event.code;
+    const button = document.querySelector(`.${codeKey}`);
+
+    if (button) {
+      event.preventDefault();
+
+      if (button.dataset.button !== 'CapsLock') {
+        if (button.classList.contains('active')) {
+          button.classList.remove('active');
+        }
+      }
+
+      if ((button.dataset.button === 'AltRight') || (button.dataset.button === 'AltLeft')) {
+        this.alt = false;
+      }
+
+      if ((button.dataset.button === 'ShiftRight') || (button.dataset.button === 'ShiftLeft')) {
+        this.shift = false;
+      }
+
+      if ((button.dataset.button === 'AltRight') || (button.dataset.button === 'AltLeft')) {
+        this.alt = false;
+      }
+
+      this.changeWordCase();
+    }
+  }
+
+  changeWordCase() {
+    const q = document.querySelectorAll(`.${this.lang} span`);
+
+    if (!this.capsLock && !this.shift) {
+      q.forEach((item) => {
+        if (item.classList.contains('noShift')) {
+          item.classList.remove('hidden');
+        } else if (!item.classList.contains('hidden')) {
+          item.classList.add('hidden');
+        }
+      });
+    } else if (!this.capsLock && this.shift) {
+      q.forEach((item) => {
+        if (item.classList.contains('isShift')) {
+          item.classList.remove('hidden');
+        } else if (!item.classList.contains('hidden')) {
+          item.classList.add('hidden');
+        }
+      });
+    } else if (this.capsLock && !this.shift) {
+      q.forEach((item) => {
+        if (item.classList.contains('caps')) {
+          item.classList.remove('hidden');
+        } else if (!item.classList.contains('hidden')) {
+          item.classList.add('hidden');
+        }
+      });
+    } else if (this.capsLock && this.shift) {
+      q.forEach((item) => {
+        if (item.classList.contains('shiftCaps')) {
+          item.classList.remove('hidden');
+        } else if (!item.classList.contains('hidden')) {
+          item.classList.add('hidden');
+        }
+      });
+    }
+  }
+

```

Comment:
can be improved (see https://github.com/KarpusKonstantin/virtual-keyboard/pull/2#discussion_r408662034)